File: py/0001.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}
        for (idx, item) in enumerate(nums):
            if target-item in table:
                return table[target-item], idx
            table[item] = idx
        return [-1, -1]

File: cpp/0001.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        map<int, int> table;
        int n = nums.size();
        for (int i = 0; i < n; i++)
        {
            if (table.count(target - nums[i]))
            {
                return {table[target - nums[i]], i};
            }
            table[nums[i]] = i;
        }
        return {1, -1};
    }
};